sympy/sets/sets.py

```diff
@@ -1605,6 +1705,9 @@ def as_relational(self, symbol):
 
         return And(A_rel, B_rel)
 
+    def _kind(self):
+        return SetKind(self.args[0].args[0].kind)
```

Comment:
This should be `return self.args[0].kind`.

We can't just presume that the first arg of a set gives the kind of its elements:
```python
In [28]: s = ImageSet(Lambda(n, n ** 2), {1, 2, 3}) - FiniteSet(3)

In [29]: s
Out[29]: 
⎧ 2 │              ⎫      
⎨n  │ n ∊ {1, 2, 3}⎬ \ {3}
⎩   │              ⎭      

In [30]: s.kind
Out[30]: SetKind(UndefinedKind)

In [31]: s.doit()
Out[31]: {1, 4, 9}

In [32]: s.doit().kind
Out[32]: SetKind(NumberKind)
```

Reply:
For Complement Class
```python
def _kind(self):
        def _measurement(x):
            if x is UniversalSet or x is S.UniversalSet:
                return float('inf')
            try:
                return len(x)
            except AttributeError:
                return int(x._measure)

        def _ele(x):
            try:
                return x.kind
            except AttributeError:
                return SetKind(NumberKind)
        if _measurement(self.args[0]) < _measurement(self.args[1]):
            return SetKind()
        elif _measurement(self.args[0]) > _measurement(self.args[1]):
            return _ele(self.args[0])
```
`try` and `except` is for in-build `set` class (ex = {1, 2, 3}) of python and this `_kind` function also accommodate for `evaluate=False` scenario

Reply:
As for `ImageSet`, will this do?
```pyhon
    def _kind(self):
        return self.doit().kind
```